src/composables/useReportDownload.ts

```diff
@@ -1,108 +1,54 @@
-import jsonexport from 'jsonexport/dist';
 import pkg from '@/../package.json';
-import { getProposalVotes, getProposal } from '@/helpers/snapshot';
-import { Vote } from '@/helpers/interfaces';
 
 export function useReportDownload() {
   const isDownloadingVotes = ref(false);
-  const downloadProgress = ref(0);
-
-  async function getCsvFile(
-    data: any[] | Record<string, any>,
-    headers: string[],
-    fileName: string
-  ) {
-    const csv = await jsonexport(data, { headers });
-    const link = document.createElement('a');
-    link.setAttribute('href', `data:text/csv;charset=utf-8,${csv}`);
-    link.setAttribute('download', `${fileName}.csv`);
-    document.body.appendChild(link);
-    link.click();
+  const errorCode: globalThis.Ref<null | Error> = ref(null);
+
+  async function downloadFile(blob: Blob, fileName: string) {
+    const href = URL.createObjectURL(blob);
+    const a = Object.assign(document.createElement('a'), {
+      href,
+      style: 'display:none',
+      download: fileName
+    });
+    document.body.appendChild(a);
+    a.click();
+    URL.revokeObjectURL(href);
+    a.remove();
   }
 
-  async function getAllVotes(
-    proposalId: string,
-    space: string,
-    totalVotesCount: number
-  ) {
-    let votes: Vote[] = [];
-    let page = 0;
-    let createdPivot = 0;
-    const pageSize = 1000;
-    let resultsSize = 0;
-    const maxPage = 5;
-    do {
-      let newVotes = await getProposalVotes(proposalId, {
-        first: pageSize,
-        skip: page * pageSize,
-        space: space,
-        created_gte: createdPivot,
-        orderBy: 'created',
-        orderDirection: 'asc'
-      });
-      resultsSize = newVotes.length;
-
-      if (page === 0 && createdPivot > 0) {
-        const existingIpfs = votes.slice(-pageSize).map(vote => vote.ipfs);
-
-        newVotes = newVotes.filter(vote => {
-          return !existingIpfs.includes(vote.ipfs);
-        });
-      }
+  async function downloadVotes(proposalId: string) {
+    isDownloadingVotes.value = true;
+    errorCode.value = null;
 
-      if (page === maxPage) {
-        page = 0;
-        createdPivot = newVotes[newVotes.length - 1].created;
-      } else {
-        page++;
+    return fetch(
```

Comment:
We should use async await:

```js
async function downloadVotes(proposalId: string) {
    isDownloadingVotes.value = true;
    errorCode.value = null;

    try {
      const response = await fetch(
        `${import.meta.env.VITE_SIDEKICK_URL}/api/votes/${proposalId}`,
        {
          method: "POST",
        }
      );

      if (!response.ok) {
        const resJson = await response.json();
        throw new Error(
          resJson.error
            ? resJson.error.message
            : `Error: received ${response.status} status code from server`
        );
      }

      const blob = await response.blob();
      downloadFile(blob, `${pkg.name}-report-${proposalId}`);
      return true;
    } catch (e) {
      if (e instanceof Error) {
        state.errorCode = new Error(`Network error: ${e.message}`);
      } else {
        state.errorCode = e;
      }
      return false;
    } finally {
      state.isDownloadingVotes = false;
    }
  }
  ```

Reply:
What do you think about using `useFetch` (https://vueuse.org/core/useFetch/#usefetch) here ?

Reply:
> What do you think about using `useFetch` (https://vueuse.org/core/useFetch/#usefetch) here ?

Sure, I've never used it but looks useful